kuflow-samples-temporal-uivision-spring/src/main/java/com/kuflow/samples/temporal/worker/uivision/TemporalConfiguration.java

```diff
@@ -0,0 +1,127 @@
+ * THE SOFTWARE.
+ */
+package com.kuflow.samples.temporal.worker.uivision;
+
+import com.kuflow.rest.KuFlowRestClient;
+import com.kuflow.samples.temporal.worker.uivision.SampleEngineWorkerUiVisionProperties.TemporalProperties.MutualTlsProperties;
+import com.kuflow.temporal.common.authorization.KuFlowAuthorizationTokenSupplier;
+import com.kuflow.temporal.common.payload.codec.EncryptionPayloadCodec;
+import com.kuflow.temporal.common.payload.codec.encryption.PayloadEncryptor;
+import com.kuflow.temporal.common.payload.codec.encryption.PayloadEncryptors;
+import com.kuflow.temporal.common.payload.codec.store.SecretStore;
+import com.kuflow.temporal.common.payload.codec.store.SecretStores;
+import com.kuflow.temporal.common.ssl.SslContextBuilder;
+import com.kuflow.temporal.common.tracing.MDCContextPropagator;
+import io.grpc.netty.shaded.io.netty.handler.ssl.SslContext;
+import io.temporal.authorization.AuthorizationGrpcMetadataProvider;
+import io.temporal.client.ActivityCompletionClient;
+import io.temporal.client.WorkflowClient;
+import io.temporal.client.WorkflowClientOptions;
+import io.temporal.common.converter.CodecDataConverter;
+import io.temporal.common.converter.DefaultDataConverter;
+import io.temporal.serviceclient.WorkflowServiceStubs;
+import io.temporal.serviceclient.WorkflowServiceStubsOptions;
+import io.temporal.serviceclient.WorkflowServiceStubsOptions.Builder;
+import io.temporal.worker.WorkerFactory;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration(proxyBeanMethods = false)
+public class TemporalConfiguration {
+
+    private final SampleEngineWorkerUiVisionProperties sampleEngineWorkerUiVisionProperties;
+
+    private final KuFlowRestClient kuFlowRestClient;
+
+    public TemporalConfiguration(
+        SampleEngineWorkerUiVisionProperties sampleEngineWorkerUiVisionProperties,
+        KuFlowRestClient kuFlowRestClient
+    ) {
+        this.sampleEngineWorkerUiVisionProperties = sampleEngineWorkerUiVisionProperties;
+        this.kuFlowRestClient = kuFlowRestClient;
+    }
+
+    @Bean(destroyMethod = "shutdown")
+    public WorkflowServiceStubs workflowServiceStubs() {
+        Builder builder = WorkflowServiceStubsOptions.newBuilder();
+        builder.setTarget(this.sampleEngineWorkerUiVisionProperties.getTemporal().getTarget());
+        builder.setSslContext(this.createSslContext());
+        builder.addGrpcMetadataProvider(new AuthorizationGrpcMetadataProvider(new KuFlowAuthorizationTokenSupplier(this.kuFlowRestClient)));
+
+        WorkflowServiceStubsOptions options = builder.validateAndBuildWithDefaults();
+
+        return WorkflowServiceStubs.newServiceStubs(options);
+    }
+
+    @Bean
+    public EncryptionPayloadCodec encryptionPayloadCodec() {
```

Comment:
I don't add this bean, to avoid mixing too many concepts here. Maybe we must create a new example explaining how to add encryption to temporal.

Reply:
See that: https://github.com/kuflow/kuflow-samples-java/blob/ad432beaf4684d45040dab419062048bf8aca4a6/kuflow-samples-temporal-loan/src/main/java/com/kuflow/samples/temporal/worker/email/SampleEngineWorkerLoan.java#L177

Reply:
yes, maybe we must delete it from there too.